cognite/client/data_classes/data_modeling/ids.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from dataclasses import asdict, dataclass, field
+from typing import ClassVar, Optional, Sequence, Tuple, Type, TypeVar, Union, cast
+
+from cognite.client.utils._auxiliary import rename_and_exclude_keys
+from cognite.client.utils._identifier import DataModelingIdentifier, DataModelingIdentifierSequence
+from cognite.client.utils._text import convert_all_keys_recursive, convert_all_keys_to_snake_case
+
+
+@dataclass
+class DataModelingId:
+    _type: ClassVar[str] = field(init=False)
+    space: str
+    external_id: str
+
+    def as_tuple(self) -> tuple[str, str]:
+        return self.space, self.external_id
+
+    def dump(self, camel_case: bool = False, include_type: bool = True) -> dict[str, str]:
+        output = asdict(self)
+        if include_type:
+            output["type"] = self._type
+        return convert_all_keys_recursive(output, camel_case)
+
+    @classmethod
+    def load(cls: Type[T_DataModelingId], data: dict) -> T_DataModelingId:
+        return cls(**convert_all_keys_to_snake_case(rename_and_exclude_keys(data, exclude={"type"})))
+
+
+T_DataModelingId = TypeVar("T_DataModelingId", bound=DataModelingId)
+
+
+@dataclass
+class VersionedDataModelingId:
+    _type: ClassVar[str] = field(init=False)
+    space: str
+    external_id: str
+    version: Optional[str] = None
+
+    def as_tuple(self) -> tuple[str, str, Optional[str]]:
+        return self.space, self.external_id, self.version
+
+    def dump(self, camel_case: bool = False, include_type: bool = True) -> dict[str, str]:
+        output = asdict(self)
+        if include_type:
+            output["type"] = self._type
+        return convert_all_keys_recursive(output, camel_case)
+
+    @classmethod
+    def load(cls: Type[T_Versioned_DataModeling_Id], data: dict) -> T_Versioned_DataModeling_Id:
+        return cls(**convert_all_keys_to_snake_case(rename_and_exclude_keys(data, exclude={"type"})))
+
+
+T_Versioned_DataModeling_Id = TypeVar("T_Versioned_DataModeling_Id", bound=VersionedDataModelingId)
+
+
+@dataclass
+class ContainerId(DataModelingId):
+    _type = "container"
+
+    @classmethod
+    def from_tuple(cls, data: tuple[str, str] | ContainerId) -> ContainerId:
```

Comment:
Don't like a "from_tuple" method accepting something which isn't a tuple. Perhaps just move this into load and have 3 branches there? So it becomes
```
def load(cls: Type[T_Versioned_DataModeling_Id], data: dict | tuple | T_Versioned_DataModeling_Id) -> T_Versioned_DataModeling_Id:
```